pubsublite/pubsublite.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+
+package pubsublite
+
+import (
+	"context"
+
+	"google.golang.org/api/option"
+
+	vkit "cloud.google.com/go/pubsublite/apiv1"
+	pb "google.golang.org/genproto/googleapis/cloud/pubsublite/v1"
+)
+
+// Client provides admin operations for Google Pub/Sub Lite resources within a
+// Google Cloud region. A Client may be shared by multiple goroutines.
+type Client struct {
+	// The Google Cloud region that this Client manages resources for.
+	Region string
+	admin  *vkit.AdminClient
+}
+
+// NewAdminClient creates a new Cloud Pub/Sub Lite client to perform admin
+// operations for resources within a given region.
+// See https://cloud.google.com/pubsub/lite/docs/locations for the list of
+// regions and zones where Google Pub/Sub Lite is available.
+func NewAdminClient(ctx context.Context, region string, opts ...option.ClientOption) (c *Client, err error) {
+	if err := ValidateRegion(region); err != nil {
+		return nil, err
+	}
+	options := []option.ClientOption{option.WithEndpoint(region + "-pubsublite.googleapis.com:443")}
+	admin, err := vkit.NewAdminClient(ctx, options...)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{Region: region, admin: admin}, nil
+}
+
+// CreateTopic creates a new topic from the given config.
+func (c *Client) CreateTopic(ctx context.Context, config *TopicConfig) (*TopicConfig, error) {
```

Comment:
Are nil configs valid for these admin operations? Should this be passed by value?

Reply:
Yes, they are required. Changed to value.